eventservice/api/utils/group_auth.py

```diff
@@ -1,18 +1,19 @@
 import controllers.configs as cfg
 import requests
+import logging
 from flask import g
 
+logger = logging.getLogger(__name__)
+
 def get_group_ids():
     group_ids = list()
     include_private_events = False
-    # user UserAuth request
-    if 'user_token_data' in g:
+    if 'user_token' in g:
         include_private_events = True
-        auth_resp = g.user_token_data
-        uin = auth_resp.get('uiucedu_uin')
-        url = "%s%s/groups" % (cfg.GROUPS_BUILDING_BLOCK_ENDPOINT, uin)
+        id_token = g.user_token
+        url = cfg.GROUPS_BUILDING_BLOCK_HOST + "/api/user/group-memberships"
         headers = {"Content-Type": "application/json",
-                  "ROKWIRE_GS_API_KEY": cfg.ROKWIRE_GROUPS_API_KEY}
+                    "Authorization": "Bearer " + id_token}
```

Comment:
Hi @sandeep-ps @bingzhang, I wanted to give a bit more context for these changes here. The issue is that we want to be able to support auth types other than Shibboleth for private group events. The Groups BB already knows how to parse all possible login types for the Core tokens and also accepts and understands how to handle Shibboleth tokens. Rather than trying to duplicate that complicated work, we thought it would be best to simply forward the user tokens to a new Groups BB endpoint protected by user token auth that parses the user ID from the token and returns the same data as the old endpoint. Please let us know if either of you have any questions or concerns about this approach. Thanks!